gui/splash_hook.py

```diff
@@ -0,0 +1,55 @@
+from mozregression import version
+import multiprocessing as mp
+import os
+import subprocess
+
+
+class Splash:
+    # Milliseconds to pass before re-checking open files.
+    CHECK_MILLISECONDS = 3000
+
+    def __init__(self):
+        # Must import here to avoid multiprocess forking issue with Core Foundation.
+        import tkinter
+
+        # Get parent PID which will be the one extracting files, etc...
+        self.ppid = str(os.getppid())
+
+        # Variable to keep track of last lsof output.
+        self.last_lsof = None
+
+        # Create the splash screen.
+        self.win = tkinter.Tk()
+        self.win.overrideredirect(True)
+        tkinter.Label(self.win, text=f"mozregression {version} starting, please wait...").pack(
+            pady=20, padx=20
+        )
+        self.win.eval("tk::PlaceWindow . center")
+        self.win.after(self.CHECK_MILLISECONDS, self.check_doneness)
+        self.win.mainloop()
+
+    def check_doneness(self):
+        """Check whether the bootloader has finished loading files."""
+
+        # NOTE: This is a bit hacky. The output from lsof is checked every few seconds
+        # and if it has not changed between those two checks, assume extraction is
```

Comment:
Should probably set a max timeout here, in case something wonky causes lsof to continuously provide different output.

Reply:
No longer needed, since we are now checking only the extracted folder.